src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -307,5 +307,12 @@ trait MarathonConf
     noshort = true,
     hidden = true,
     default = Some(MesosHeartbeatMonitor.DEFAULT_HEARTBEAT_FAILURE_THRESHOLD))
+
+  lazy val drainingTime = opt[Long](
+    "draining_time",
```

Comment:
Does the name "draining time" imply that the parameter does more than it actually does? When I think of "draining", I would have the impression that Marathon would be killing the tasks and moving them to other machines.

Also, have we considered whether this makes more sense as a per-app setting, vs globally? It seems some tasks may be much more tolerant to failure than others, and would prefer to get as many resources as available. IE some background compute worker?

It is awesome that there is an implementation that is useful. Would it be helpful to take a step back and think of this from a macro perspective? Perhaps this change is on the road to where we ultimately want to go, I'd just like to be sure.